File: src/main/java/com/example/storage/controller/dto/IssueCreateDto.java
package com.example.storage.controller.dto;

import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class IssueCreateDto {
    @Positive
    private long equipmentId;
    @NotNull
    private LocalDate returnDate;
}

File: src/main/kotlin/ru/nto/storage/controller/dto/IssueCreateDto.kt
package ru.nto.storage.controller.dto

import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Positive
import java.time.LocalDate

data class IssueCreateDto(
    @field:Positive
    val equipmentId: Long,
    @field:NotNull
    val returnDate: LocalDate
)
